api/docker.ts

```diff
@@ -97,8 +121,217 @@ async function sizeHandler ({ scope, name, tag, architecture, variant }: PathArg
   const sizeInMegabytes = (imageData.size / 1024 / 1024).toFixed(2)
 
   return {
-    subject: 'docker image size',
+    subject: 'docker size',
     status: `${sizeInMegabytes} MB`,
     color: 'blue'
   }
 }
+
+async function layersHandler ({ scope, name, tag, architecture, variant }: PathArgs) {
+  tag = tag ? tag : 'latest'
+  architecture = architecture ? architecture : 'amd64'
+  variant = variant ? variant : ''
+
+  const token = (await getDockerAuthToken(scope, name)).token
+  
+  if (! token) {
+    return {
+      subject: 'docker metadata',
+      status: `unknown image: ${scope}/${name}`,
+      color: 'grey'
+    }
+  }
+
+  let headers = {
+    authorization: `Bearer ${token}`,
+    accept: `application/vnd.docker.distribution.manifest.list.v2+json`
+  }
+  let path = `v2/${scope}/${name}/manifests/${tag}`
+  
+  const manifest_list = (await queryDockerRegistry(path, headers)).manifests
```

Comment:
We can put these into `libs/docker.ts`, throw a `BadgenError` for unknown image, like [github badge](https://github.com/badgen/badgen.net/blob/370a4a87b6c3278965923528b7156b0250446561/libs/github.ts#L32).

Reply:
I moved all of the token and error checking logic into libs\docker.ts with commit [1d62a11](https://github.com/badgen/badgen.net/pull/442/commits/1d62a11926ccca73f8fab8246d7c799c51702dd2)